File: algorithms/cpp/518.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int change(int amount, vector<int> &coins) {
        vector<int> dp(amount+1, 0);
        dp[0] = 1;
        for (auto coin: coins) {
            for (int i = 0; i <= amount; i++) {
                if (i-coin >= 0)
                    dp[i] += dp[i-coin];
            }
        }
        return dp[amount];
    }
};

int main() {
    return 0;
}

File: algorithms/rust/518.rs
impl Solution {
    #[allow(dead_code)]
    pub fn change(amount: i32, coins: Vec<i32>) -> i32 {
        let amount = amount as usize;
        let mut dp = vec![0; amount + 1];
        dp[0] = 1;
        for coin in coins.into_iter().map(|x| x as usize) {
            for i in coin..=amount {
                dp[i] += dp[i - coin];
            }
        }
        dp[amount]
    }
}

struct Solution;

fn main() {
    println!("Hello, world!");
}
